mopro-ffi/tests/bindings/test_mopro_keccak2.swift

```diff
@@ -1,5 +1,5 @@
-import mopro
```

Comment:
there are some errors that I haven't solved
maybe add tests in the next PR?

```bash
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:20:16: error: unresolved reference: jna
import com.sun.jna.Library
               ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:21:16: error: unresolved reference: jna
import com.sun.jna.IntegerType
               ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:22:16: error: unresolved reference: jna
import com.sun.jna.Native
               ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:23:16: error: unresolved reference: jna
import com.sun.jna.Pointer
               ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:24:16: error: unresolved reference: jna
import com.sun.jna.Structure
               ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:25:16: error: unresolved reference: jna
import com.sun.jna.Callback
               ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:26:16: error: unresolved reference: jna
import com.sun.jna.ptr.*
               ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:37:2: error: unresolved reference: Structure
@Structure.FieldOrder("capacity", "len", "data")
 ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:38:25: error: unresolved reference: Structure
open class RustBuffer : Structure() {
                        ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:41:25: error: unresolved reference: Pointer
    @JvmField var data: Pointer? = null
                        ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:43:34: error: unresolved reference: Structure
    class ByValue: RustBuffer(), Structure.ByValue
                                 ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:44:38: error: unresolved reference: Structure
    class ByReference: RustBuffer(), Structure.ByReference
                                     ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:63:13: error: unresolved reference: it
            it.order(ByteOrder.BIG_ENDIAN)
            ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:73:31: error: unresolved reference: ByReference
class RustBufferByReference : ByReference(16) {
                              ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:79:23: error: unresolved reference: getPointer
        val pointer = getPointer()
                      ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:89:23: error: unresolved reference: getPointer
        val pointer = getPointer()
                      ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:91:15: error: unresolved reference: writeField
        value.writeField("capacity", pointer.getInt(0))
              ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:92:15: error: unresolved reference: writeField
        value.writeField("len", pointer.getInt(4))
              ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:93:15: error: unresolved reference: writeField
        value.writeField("data", pointer.getPointer(8))
              ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:105:2: error: unresolved reference: Structure
@Structure.FieldOrder("len", "data")
 ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:106:27: error: unresolved reference: Structure
open class ForeignBytes : Structure() {
                          ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:108:25: error: unresolved reference: Pointer
    @JvmField var data: Pointer? = null
                        ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:110:37: error: unresolved reference: Structure
    class ByValue : ForeignBytes(), Structure.ByValue
                                    ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:149:17: error: unresolved reference: it
                it.order(ByteOrder.BIG_ENDIAN)
                ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:152:18: error: unresolved reference: writeField
            rbuf.writeField("len", bbuf.position())
                 ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:186:2: error: unresolved reference: Structure
@Structure.FieldOrder("code", "error_buf")
 ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:187:38: error: unresolved reference: Structure
internal open class RustCallStatus : Structure() {
                                     ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:191:38: error: unresolved reference: Structure
    class ByValue: RustCallStatus(), Structure.ByValue
                                     ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:259:39: error: unresolved reference: IntegerType
public class USize(value: Long = 0) : IntegerType(Native.SIZE_T_SIZE, value, true) {
                                      ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:259:51: error: unresolved reference: Native
public class USize(value: Long = 0) : IntegerType(Native.SIZE_T_SIZE, value, true) {
                                                  ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:261:5: error: 'toByte' overrides nothing
    override fun toByte() = toInt().toByte()
    ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:261:29: error: unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
public inline fun String.toInt(): Int defined in kotlin.text
public inline fun String.toInt(radix: Int): Int defined in kotlin.text
private fun Iterable<FlagEnum>.toInt(): Int defined in kotlin.text
    override fun toByte() = toInt().toByte()
                            ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:262:5: error: 'toChar' overrides nothing
    override fun toChar() = toInt().toChar()
    ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:262:29: error: unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
public inline fun String.toInt(): Int defined in kotlin.text
public inline fun String.toInt(radix: Int): Int defined in kotlin.text
private fun Iterable<FlagEnum>.toInt(): Int defined in kotlin.text
    override fun toChar() = toInt().toChar()
                            ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:263:5: error: 'toShort' overrides nothing
    override fun toShort() = toInt().toShort()
    ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:263:30: error: unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
public inline fun String.toInt(): Int defined in kotlin.text
public inline fun String.toInt(radix: Int): Int defined in kotlin.text
private fun Iterable<FlagEnum>.toInt(): Int defined in kotlin.text
    override fun toShort() = toInt().toShort()
                             ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:270:19: error: unresolved reference: Native
            when (Native.SIZE_T_SIZE) {
                  ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:271:33: error: unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
public inline fun String.toInt(): Int defined in kotlin.text
public inline fun String.toInt(radix: Int): Int defined in kotlin.text
private fun Iterable<FlagEnum>.toInt(): Int defined in kotlin.text
                4 -> buf.putInt(toInt())
                                ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:272:34: error: unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
public inline fun String.toLong(): Long defined in kotlin.text
public inline fun String.toLong(radix: Int): Long defined in kotlin.text
                8 -> buf.putLong(toLong())
                                 ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:273:72: error: unresolved reference: Native
                else -> throw RuntimeException("Invalid SIZE_T_SIZE: ${Native.SIZE_T_SIZE}")
                                                                       ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:282:21: error: unresolved reference: Native
            get() = Native.SIZE_T_SIZE
                    ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:289:30: error: unresolved reference: Native
                return when (Native.SIZE_T_SIZE) {
                             ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:292:76: error: unresolved reference: Native
                    else -> throw RuntimeException("Invalid SIZE_T_SIZE: ${Native.SIZE_T_SIZE}")
                                                                           ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:349:35: error: unresolved reference: Library
private inline fun <reified Lib : Library> loadIndirect(
                                  ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:352:12: error: unresolved reference: Native
    return Native.load<Lib>(findLibraryName(componentName), Lib::class.java)
           ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:358:33: error: unresolved reference: Library
internal interface _UniFFILib : Library {
                                ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:369:49: error: unresolved reference: Pointer
    fun uniffi_mopro_fn_free_moprocircom(`ptr`: Pointer,_uniffi_out_err: RustCallStatus, 
                                                ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:372:8: error: unresolved reference: Pointer
    ): Pointer
       ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:373:57: error: unresolved reference: Pointer
    fun uniffi_mopro_fn_method_moprocircom_setup(`ptr`: Pointer,`wasmPath`: RustBuffer.ByValue,`r1csPath`: RustBuffer.ByValue,_uniffi_out_err: RustCallStatus, 
                                                        ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:375:66: error: unresolved reference: Pointer
    fun uniffi_mopro_fn_method_moprocircom_generate_proof(`ptr`: Pointer,`circuitInputs`: RustBuffer.ByValue,_uniffi_out_err: RustCallStatus, 
                                                                 ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:377:64: error: unresolved reference: Pointer
    fun uniffi_mopro_fn_method_moprocircom_verify_proof(`ptr`: Pointer,`proof`: RustBuffer.ByValue,`publicInput`: RustBuffer.ByValue,_uniffi_out_err: RustCallStatus, 
                                                               ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:679:28: error: unresolved reference: Pointer
    protected val pointer: Pointer
                           ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:705:58: error: unresolved reference: Pointer
    internal inline fun <R> callWithPointer(block: (ptr: Pointer) -> R): R {
                                                         ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:737:14: error: unresolved reference: Pointer
    pointer: Pointer
             ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:800:70: error: unresolved reference: Pointer
public object FfiConverterTypeMoproCircom: FfiConverter<MoproCircom, Pointer> {
                                                                     ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:801:45: error: unresolved reference: Pointer
    override fun lower(value: MoproCircom): Pointer = value.callWithPointer { it }
                                            ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:803:30: error: unresolved reference: Pointer
    override fun lift(value: Pointer): MoproCircom {
                             ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:810:21: error: unresolved reference: Pointer
        return lift(Pointer(buf.getLong()))
                    ^
target/tmp/mopro-ffi-a7c28f91c8360f12/uniffi/mopro/mopro.kt:818:21: error: unresolved reference: Pointer
        buf.putLong(Pointer.nativeValue(lower(value)))
                    ^
Error: running `kotlinc` failed
test uniffi_foreign_language_testcase_test_mopro_kts ... FAILED

failures:

failures:
    uniffi_foreign_language_testcase_test_mopro_kts

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 5.54s
```

Reply:
hm interesting that it works in simulator but not as kotlin tests

follow up pr: sure, want to add as gh issue or so? or can just keep track of in https://github.com/oskarth/mopro/issues/17 with updated checklist

Reply:
we can keep track in #17 
It matches the first criteria